range_queries/sparse_table.cpp

```diff
@@ -0,0 +1,54 @@
+// Implementation of Sparse Table
+//
+// Running Time Complexity
+// Build : O(NlogN)
+// Range Query : O(1)
+
+
+#include<iostream>
+#include<algorithm>
+
+const int N = 100005;
+const int LOGN = 20;
+
+int n;                  // length of the array
+int A[N];
+int logs[N];            // logs[i] = maximum p s.t. 2^p <= i
+int table[LOGN][N];     // table[i][j] = [j, j+2^i]
+
+void computeLogs(int n) {
+    logs[1] = 0;
+    for (int i = 2 ; i < n ; i++)
+        logs[i] = logs[i/2] + 1;
+}
+
+void buildTable(int n) {
+    int i, j, curLen;
+    for (i = 0 ; i <= logs[n] ; i++) {
+        curLen = 1LL << i;
+        for (j = 0 ; j + curLen < n ; j++) {
+            if (curLen == 1)
+                table[i][j] = A[j];
+            else
+                table[i][j] = std::min(table[i-1][j], table[i-1][j + curLen/2]);
```

Comment:
Make it look like C++ code, not C code.

```suggestion
            if (curLen == 1) {
                table[i][j] = A[j];
            } else {
                table[i][j] = std::min(table[i-1][j], table[i-1][j + curLen/2]);
            }
```

Reply:
Previous lint codes were in C format, so did changes accordingly. Updated everything and pushed it.